modules/dnn/src/graph_simplifier.cpp

```diff
@@ -69,8 +69,14 @@ int Subgraph::getInputNodeId(const Ptr<ImportGraphWrapper>& net,
     const int numNodes = net->getNumNodes();
     for (int i = 0; i < numNodes; ++i)
     {
-        if (net->getNodeName(i) == name)
-            return i;
+        Ptr<ImportNodeWrapper> node = net->getNode(i);
+        const int numOutputs = node->getNumOutputs();
+        for (int j = 0; j < numOutputs; j++)
+        {
+            std::string outName = (node->getOutputName(j) != "graph_input") ? node->getOutputName(j) : net->getNodeName(i);
+            if (outName  == name)
+                return i;
+        }
```

Comment:
Please take a look at commit https://github.com/dkurt/opencv/commit/c39ea60140c247a90d8ccf43c9c9d4b25e2a0d6a

Now I see the problem. Let's check outputs names from network rather than nodes. Sorry for misleading!

Reply:
ohk, got it. I have added you suggested changes. Thank you so much.